de-webapp/src/main/java/org/iplantc/de/server/rpc/GwtRpcController.java

```diff
@@ -67,6 +67,9 @@ public String processCall(String payload) throws SerializationException {
 
             RPCRequest rpcRequest = RPC.decodeRequest(payload,
                                                       this.remoteServiceClass);
+            if (remoteService instanceof DeGwtRemoteLoggingServiceImpl) {
+                ((DeGwtRemoteLoggingServiceImpl)remoteService).setRequest(getThreadLocalRequest());
```

Comment:
Just an "educating Ian more" comment: Everything makes sense to me other than this block -- what does it do, and where does `getThreadLocalRequest()` come from? I can't find other references to it in the UI, at least.